### PR TITLE
Add disabling macros

### PR DESCRIPTION
**Pull request by thetheodor**

### CODE REVIEW

**Comment by Quarub** [comment]
https://github.com/DeadCodeProductions/dead_instrumenter/blob/d748ba961a7d365e5c45c1518184c29cf1336be2/python_src/dead_instrumenter/utils.py#L82

13 -> 14

File: src/DeadInstrumenter.cpp [hunk 1]
```diff
@@ -258,25 +536,156 @@ AST_MATCHER(DoStmt, DoAndWhileNotMacroAndInMain) {
...
-            .with(inMainAndNotMacro,
-                  hasBody(stmt(inMainAndNotMacro).bind("body")));
+        whileStmt(inMainAndNotMacro,
+                  hasBody(stmt(inMainAndNotMacro).bind("body")))
+            .bind("loop");
+    auto macroActions = flatten(
+        addDeleteMacroPre(changeTo(whileBegin("loop"), cat("while"))),
+        insertAfter(LParenLoc("loop"), cat("\n\n#endif\n\n")),
+        addDeleteMacro(RParenLoc("loop"), cat("\n\n")),
+        insertAfter(RParenLoc("loop"), cat("\n\n#else\n;\n#endif\n\n")));
```
Comment (Quarub) [inline src/DeadInstrumenter.cpp:681]: Potentially dangerous non-double `\n`? Or is this just problematic on the boundaries to "real code"?

**Comment by Quarub** [comment]
I thought of including the binary in the package directly. With the option to build it yourself.
When installing with pip, you can specify additionally for installation. For example,  sqlalchemy has `sqlalchemy[mypy]` to also install the mypy plugin and sqlalchemy stubs.

**Comment by thetheodor** [comment]
Ah okay, I misunderstood. I'd do this in a separate PR. So maybe we merge this (after I fix CMake) and then look into distributing a binary in v0.1.1?

**Comment by Quarub** [comment]
:+1:

**Comment by thetheodor** [comment]
Updates the cmake stuff. It should be a bit smarter now

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,29 +1,42 @@
...
 find_package(Clang REQUIRED CONFIG)
 message(STATUS "Using ClangConfig.cmake in ${Clang_DIR}")
+list(APPEND CMAKE_MODULE_PATH ${CLANG_CMAKE_DIR})
+include(${CLANG_CMAKE_DIR}/AddClang.cmake)
+message(STATUS "ClangDefinitions ${CLANG_DEFINITIONS}")
 
 
-if(NOT "13" VERSION_EQUAL "${LLVM_VERSION_MAJOR}")
-  message(FATAL_ERROR "Found LLVM ${LLVM_VERSION_MAJOR}, but need LLVM 12")
+if(NOT "13" VERSION_LESS_EQUAL "${LLVM_VERSION_MAJOR}")
```
Comment (Quarub) [inline CMakeLists.txt:18]: `if("${LLVM_VERSION_MAJOR}" VERSION_LESS "13")` ?

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,29 +1,42 @@
...
+  message(FATAL_ERROR "Found LLVM ${LLVM_VERSION_MAJOR}, but need LLVM 14")
```
Comment (Quarub) [inline CMakeLists.txt:19]: `but need LLVM >=13`

Comment (thetheodor) [reply CMakeLists.txt]: fixed

**Review by Quarub — APPROVED** [review]

**Pull request merged**